APE0.rst

```diff
@@ -0,0 +1,360 @@
+The Astropy Project Governance Charter
+======================================
+
+author: The Astropy Governance Working Group; Tom Aldcroft, Matt Craig, Kelle Cruz, Lia Corrales, Steve Crawford, Nicole Foster, Pey Lian Lim, Stuart Mumford, Adrian Price-Whelan, Thomas Robitaille, David Shupe, Brigitta Sipocz, Erik Tollerud
+
+date-created: 2020 July 31
+
+date-last-revised: 2020 October 6
+
+date-accepted: <replace with accepted date>
+
+type: Process
+
+status: Discussion
+
+Abstract
+========
+This APE defines the formal fundamental governance process for the Astropy
+Project. The Astropy Project broadly follows the philosophy of do-ocracy:
+project participants who gain experience and knowledge through sustained
+community contributions and have demonstrated adherence to core principles of
+inclusion and consensus-building will be entrusted with increasing levels of
+authority. This peaks with the Coordination Committee, which has relatively
+broad authority but seeks to exercise it as rarely as possible by working
+towards consensus. This is because at its core, the Project is both by and for
+its community, and therefore the community is the final source of authority and
+direction.
+
+Detailed Description
+====================
+The Governance Charter is described in this section. The initial version is the
+outcome of several months of discussion by the Astropy Governance Working Group,
+which aimed to both codify existing informal governance rules as well as update
+them with new ideas that are consistent with the philosophy of and vision for
+the Astropy project.
+
+Ultimate Authority: The Astropy Community
+-----------------------------------------
+The Astropy Project derives its authority from the whole community - developers,
+users, and even potential participants in the broader astronomical and
+scientific software community. All have a vested interest in the success of the
+Project, so all governance in the end belongs to this community in its entirety.
+The governance Charter described below provides levels of delegation of that
+authority, particularly to the Voting Members and the Coordination Committee.
+
+The Coordination Committee
+--------------------------
+
+Composition
+^^^^^^^^^^^
+The Coordination Committee is a committee that is composed of four members. To
+represent a broad set of interests, the Coordination Committee should have no
+more than two members from a particular employer. The Ombudsperson (described in
```

Comment:
Since there is a clause that a strict majority is needed, i.e., 3 out of 4, I guess 2 from one place cannot really drive decision making, though they could stall it. But it would seem nice if CoCo had geographic breadth, and having 5 members would have the advantage that perhaps 4 could be considered a quorum so that one doesn't always have to get everyone together.

The main question may be how many people are going to want to be nominated? I suspect that we'll regularly have elections by acclaim or with a choice of N out of N+1... In that respect, one option would be to say that a second member from a given institution can only happen if otherwise there would be too few candidates.

Anyway, I'd also support making it 5. Certainly, 3 seems too few (one forceful individual can sway another and decide too much), and 6 too many (no-one will feel responsible to actually get things done).

Reply:
👍 on five members for all the reasons @mhvk listed.

Reply:
![image](https://user-images.githubusercontent.com/314716/95623331-fd754980-0a6c-11eb-89ab-05350fe8ffd3.gif)

Just for the record I would be happy with five which as a side benefit **guarantees** at least one new person on the CoCo at the first elections.

Reply:
>if otherwise there would be too few candidates

Could you clarify this? In my mind, multiple people can still run from a given institute during an election (and I would consider it even a good feature as it would be up to the community to decide whom they prefer not some internal politics). However it would be only the one with the higher votes would serve on the committee (basically we have the same situation already with the current draft when 3 people are nominated from the same place)

Reply:
Didn't we debate for a lot time on the numbers `{3, 4, 5}` and then settled on 4 somehow. Are we doing it all over again?

Reply:
we can go into the particulars of how that close vote came to be, but the point here really is that 2 out of the 4 is disproportional if it ends up in the same building.

Reply:
@bsipocz - sorry, I was unclear. I just meant that if there are really few candidates, it would be better to have more than one from one institute than fewer than the nominal number of CoCo members. So, with multiple candidates from a single institute, the one with the most votes would get in, and any others only if there were no other candidates with votes left.

Reply:
Note that the 4 vs 5 question  was discussed in the WG - most everyone agreed 3 is too few and 6 is too many. When we did a straw poll for 4 vs 5 it was a 3-to-4 vote and no one objected to either, so this was just only chosen by a slim majority. 

 But I think I was the prime advocate for 4 so I'll try to give the argument (I will try to state my position in a way that does not seem heavy-handed, @bsipocz, but please let me know either here or via Slack DM or through an intermediary if there's some way I can state this position in a way you find less concerning.) Currently we have 4 members in the CoCo, which requires 3/4 of the committee to agree to be able to approve something. That is *good* because it means if more than one person objects to something we have to revise the idea until there's only one (or typically zero) outliers. With 5 people, there's much more potential for there to be a "minority faction" of 2 that feels marginalized.  4 prevents that while still having a fairly broad perspective. The current CoCo (and according to the notes, at least some fraction of the WG) agreed that the better solution to the CoCo being overwhelmed with tasks is for them to delegate more rather than grow to a larger size.

On the two-from-one-place question, indeed @mhvk's point is what motivated the discussion there - that 2 is a number that at a size of 4 cannot control the agenda. And if an institution is getting two people on the CoCo solely to stall progress in Astropy, well... I think we have bigger problems. But the added part of the two-for-one-place discussion is that there's some concern that it's going to be difficult to find enough candidates (which growing to 5 magnifies), so adding a further restriction makes that even harder.

Reply:
Now for my perspective just as *me*: I like 4 for the reasons I said above.  But I can live with 5 if others are not persuaded by it, and similar for the two-from-one-place vs one-from-one-place.


Although really I have to say, @astrofrog's point in https://user-images.githubusercontent.com/314716/95623331-fd754980-0a6c-11eb-89ab-05350fe8ffd3.gif may be the most persuasive...

Reply:
>  that 2 is a number that at a size of 4 cannot control the agenda

2 totally controls the agenda, it's even enough that 1 of the remaining two doesn't care that much about that particular issue to put up an argument. There is a clear desire (see @mhvk's comment above, or the outcome of preferences on the anonimoous sticky notes last year's coordination meeting) to grow geographically and the current 3/4 of the coco and 3/4 of the finance residing in the NE corridor could be made worse only by having multiple people from the same institute. 

I can live with the 4, but still don't see in the good thing a setup where you everywhere argues for consensus yet prefer a 3 vs 1  stand than working for a middle ground solution with a "minority fraction". If it's always the same 2 ends up being the minority that's a really good indication that something is not going exactly well. 

If the delegation acually works out it may be indeed a way to offload the load, but you have fighted all the way so far for each delegation item (the long discussion about keeping the decision with the coco about the affiliated review comes to mind for example).

Reply:
FWIW, I also prefer 4 over 5 on the CoCo. I think we, the current CoCo, have put a lot of effort into delegating, offloading, putting things to the community to work on and that we should continue to improve on that front instead of adding more people to the CoCo.

I also don't mind if the max is 1 per institution.

Since we may not easily find consensus on these two issues, is it worth figuring out how to do straw polls at this stage? I would be happy with just putting the question in the #project slack channel and letting people emoji vote.

Reply:
@bsipocz original comment was mainly about not allowing 2 people per institution, and I definitely think input from the current CoCo is super relevant.

How about making the sentence clearer that it would only be allowed if otherwise there would not be enough people? E.g., how about the following (also replacing "broad set of interests" with "diversity" as that can include more than just interest, including geography; could be "diversity and broad set of interests" too):
```
To represent the diversity of the astropy community, the Coordination Committee
should not have more than one member from any particular employer
(unless otherwise it is not possible to get full membership).
```

Reply:
👍 to @mhvk's suggested wording although 'full membership' might have to be spelt out more clearly

Reply:
I don't have an opinion on different numbers, but wouldn't it make sense to require--or at least prefer (since it may not always be possible depending on who volunteers) that at least one member of the committee have software engineering experience?  E.g. I don't think it should all be people in strictly academic positions.

Reply:
I like @embray 's thought. Similarly, we might want to state that no more than 2 (or 3 for a five member committee) can live on the same continent (I suggest the following lists of continents: Europe, North America, Middle and South America, Africa, Asia, Australia + New Zeeland + Antartica).

Other axes of diversity might be more difficult( e.g. it would be great to have different socio-economic backgrounds, but I don't want anyone to reveal their parents income just to be able to volunteer for a committee).

The real problem with making this a requirement is what to do if we don't have enough candidates, e.g. if we need to fill a seat with a non-US person and we only have US volunteers. If instead we only call it a "preference", it might not help much in practice: In that case, anyone can volunteer for any seat, and it's up to the voters to choose - and they may or may not follow that "preference".

Reply:
Also like it, but share the fear that in practice there will be few candidates. One big advantage of the staggered replacement scheme proposed here is that one can quickly adjust if the committee lacks diversity along any given axis - i.e., twist people's arms so that they will allow themselves to stand for the next election...

Anyway, maybe it is best for now to express a wish for diversity in the broadest sense and see how it goes?

Reply:
I feel @embray's point has more practical consequences. E.g. if the CoCo swifts away from the development process we can easily end up again in the same situation as it was for this past feature freeze where they make decisions overriding the developers without giving any thought about the consequences of those decisions on the software (release process in this case), then conveniently putting the blame back on the one who had no say in the matters, like things happening arbitrarily in a vacuum.

And there are examples in the broader python community where that is exactly what is happening, those in the top committees has nothing to do with the development process any more resulting developers are burning out in an extreme rate as they are the ones having to deal decisions (at least the ones I was talking to, all mentioned this as a big issue within their projects).

Reply:
While I definitely agree with @embray in principle, in practice I'm with @mhvk and @hamogu that it's going to be very difficult to *require* any of these things.  

The added problem is that if these are elections, then it's really up to the voting membership to decide. Do we actually say "person X got half as many votes as person Y but since Y is a software engineer they get the seat"?  Or do we just leave that seat open?

On top of that, in reality it's very hard to define things like "software engineering experience".  Consider someone who has an astronomy PhD but spent several years at a software company in an "engineering" role - do they count?  What about someone who got an CS/SE undergraduate degree but then got an astronomy PhD, and has been doing astronomy research for 10 years after that.  Do they count as an "engineer".  In practice I think all of these things are too hard to put down as requirements, but are much easier to state as *desirable goals*.

So I would propose wording like @mhvk's but bringing in more of these ideas, like:
```
The Coordination Committee should aim to represent the diversity of the astropy 
community.  Hence it should aim to have geographic diversity, a diversity of
astronomy and developer expertise, and aim to not have more than one 
member from any particular employer. While this may not be practical given
the realities of any given election, the community should work in this direction.
```

Reply:
(although if someone feels especially strongly that the "more than one from any institution" is *more* important, that can always be strengthened)

Reply:
What is even diversity of developer expertise mean? Someone with no experience whatsoever should be on coco for the sake of "developer expertise" diversity, setting the roadmap, timelines, etc? 
After all, professional software teams are managed by people who have developer side/technical experience, I fail to see why it should be different here. Certainly, we need more people who are not developers, but they should not be the ones on the top committee.

And I would define SE in this context to be someone who is working as an engineer, independently of their qualifications. We certainly need someone on the committee who represent those who are full time doing engineering, who need to deliver software for a timeline, who has engineering background knowledge, best practices, etc, rather than whose focus is primarily academic (papers, teaching, telescope operations). So, yes, in my view @astrofrog, @pllim, and @embray are engineers. Many others not so much.

Also, @mhvk's phrasing for the employer cause is much stronger, and thus I would prefer it: 
https://github.com/astropy/astropy-APEs/pull/61#discussion_r502715141

Reply:
@eteq - I think I would strengthen the one-from-any-institute by perhaps writing "Hence, it should have diversity in geography, skill set (astronomy vs developer), background, etc. Furthermore, unless it would be short of candidates otherwise, it should not have more than one member from any particular employer."

Note that this is also where the voting system is somewhat important. For instance, suppose there are 4 candidates for 2 positions, with 2 from a given institution, and I think those 2 are the best but I would want my vote to indicate that it should be one of those plus another? I guess one would need to rank candidate teams (since we won't ever have more than 4, this is do-able...).

Reply:
I agree in the sentiment of diversity along many axes, including employer and continent. But I am a little concerned that  it will require making many micro decisions to adopt these as strict requirements. For example, how do we define "employer"? If @astrofrog takes a subcontract from STScI, would that be considered his employer? Or for @kelle, who may be funded by the Flatiron Institute some semesters?

Reply:
Sitting in the same building clearly falls in the 2 from one place category. As for subcontracting, I would not count, just as I would not count Kelle's company getting the ST grant/subcontract (but would count e.g two owners of the same software company). 

Also, we have the same question with the current draft but not for 2 but 3.

Reply:
Or an employer/role/affiliation gets into your email signature, etc then it's important enough to count for this conflict of interest situation.

Reply:
> Sitting in the same building clearly falls in the 2 from one place category. 

Hmm, I see that, but then again there are people in the software engineering team that I work in that work closely on the same team but are both not in the same building and indeed not even the same state.

So with that as an I think I'm hearing consensus at least that the "from the same employer" clause should be strengthened to something like @mhvk's recent suggestion, which I'm fine with. 

However, this discussion seems to me to be revealing that it's *very* difficult to give a clear unambiguous definition of what "employer" means in this context. I still don't object to the wording, but I think it means we shouldn't spend a ton more time on the precise definition unless someone wants to offer a compact and concrete suggestion for the text itself.

Reply:
On the other thread:

> What is even diversity of developer expertise mean? Someone with no experience whatsoever should be on coco for the sake of "developer expertise" diversity, setting the roadmap, timelines, etc?

I think @mhvk's wording might have clarified this better that it's the diversity of skill sets that's relevant. The CoCo's role currently includes a mix of jobs that range from more infrastructure-focused, to specific feature-development, to outreach to the astronomy community, to writing grant proposals, to user support. That could change over time (as discussed at the last coordination meeting, delegating more of that to interested sub-groups is hopefully the next governance stage after APE0), but since all of those are integral parts of the project, it seems legit that the CoCo should represent all of that where possible.  But it's also really hard to *mandate* that since a lot of it is subjective, hence my attempt to say these are goals not hard requirements.

> After all, professional software teams are managed by people who have developer side/technical experience, I fail to see why it should be different here. Certainly, we need more people who are not developers, but they should not be the ones on the top committee.

This is not a true statement in general. There's a constantly-raging debate in the software industry about whether it's better to have professional managers, technical managers, or a mix of the two. FOSS, if anything, is unusual in having mostly technical types at the very top.  But given my statement above, I think the reality, as in so many things, is that it's better to have a breadth of perspective as long as they can work efficiently together.

Reply:
Does that clear the idea up here?  And if so, I think this would be my new proposed wording ( including some from @mhvk, some from my suggestion, and some from the original text):

```
The Coordination Committee should aim to represent the diversity of the Astropy 
community.  Hence, it should aim for diversity in geography, skill set (both 
astronomy and developer), background, etc. Furthermore, unless it would 
be short of candidates otherwise, it should not have more than one member from 
any particular employer."
```

Reply:
I have just thought of one reason to make it 5. It would guarantee that there would be at least one new person on the first elected CoCo. Personally, I think this would be a good thing.

Reply:
Note that in the version I'm about to push up, I've updated the paragraph roughly as  I said above, but have not done anything yet on the 4 vs 5 question

Reply:
Following a comment by @hamogu lower down, I made it so the APE now says four or five with the number to be decided at the initial election (see also https://github.com/astropy/astropy-APEs/pull/61#issuecomment-769284716)